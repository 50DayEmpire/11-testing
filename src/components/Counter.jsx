import React, { useState } from "react";

const Counter = () => {
  const [contador, setcontador] = useState(0);

  return (
    <>
      <h2 role="counter">Counter: {contador}</h2>
      <button onClick={() => setcontador((actual) => actual + 1)}>+1</button>
      <button
        aria-label="disminuir"
        onClick={() => setcontador((actual) => actual - 1)}
      >
        -1
      </button>
    </>
  );
};

export default Counter;
