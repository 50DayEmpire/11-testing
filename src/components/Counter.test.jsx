import { fireEvent, render, screen } from "@testing-library/react";
import Counter from "./Counter";
import userEvent from "@testing-library/user-event";
import { expect, it } from "vitest";

describe("Probar el componente <Counter/>", () => {
  beforeEach(() => {
    render(<Counter />);
  });

  it("Primer Snapshots", () => {
    expect(screen.getByRole("counter")).toMatchSnapshot();
  });

  it("Prueba: el titulo y estado se renderizan correctamente", () => {
    //Actual
    expect(screen.getByText("Counter: 0").tagName).toBe("H2");
  });

  it("Click en +1 aumenta correctamente", () => {
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByRole("counter").textContent).toContain("Counter: 1");
  });
  it("Click en -1 disminuye correctamente", async () => {
    await userEvent.click(screen.getByLabelText("disminuir"));
    expect(screen.getByRole("counter").textContent).toContain("Counter: -1");
  });
  it("Simular comportamiento de usuario", async () => {
    const btnAum = screen.getByText("+1");
    const btnDis = screen.getByLabelText("disminuir");

    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    expect(screen.getByRole("counter").textContent).toContain("Counter: 7");

    await userEvent.click(btnDis);
    await userEvent.click(btnDis);
    await userEvent.click(btnDis);
    expect(screen.getByRole("counter").textContent).toContain("Counter: 4");

    await userEvent.click(btnDis);
    await userEvent.click(btnDis);
    await userEvent.click(btnDis);
    await userEvent.click(btnDis);
    await userEvent.click(btnDis);
    expect(screen.getByRole("counter").textContent).toContain("Counter: -1");

    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    expect(screen.getByRole("counter").textContent).toContain("Counter: 1");
  });
  it("Click en Reset vuelve a 0", async () => {
    const btnAum = screen.getByText("+1");
    const btnDis = screen.getByLabelText("disminuir");
    const btnReset = screen.getByText("Reset");
    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    await userEvent.click(btnAum);
    expect(screen.getByRole("counter").textContent).toContain("Counter: 3");
    await userEvent.click(btnReset);
    expect(screen.getByRole("counter").textContent).toContain("Counter: 0");
    await userEvent.click(btnReset);
    expect(screen.getByRole("counter").textContent).toContain("Counter: 0");
    await userEvent.click(btnDis);
    await userEvent.click(btnDis);
    expect(screen.getByRole("counter").textContent).toContain("Counter: -2");
    await userEvent.click(btnReset);
    expect(screen.getByRole("counter").textContent).toContain("Counter: 0");
  });
});
