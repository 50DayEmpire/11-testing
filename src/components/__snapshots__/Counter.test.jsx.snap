// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Probar el componente <Counter/> > Primer Snapshots 1`] = `
<h2
  role="counter"
>
  Counter: 
  0
</h2>
`;
